From f3374b0c06a6e287e3ee8c68471fcab8150c2318 Mon Sep 17 00:00:00 2001
From: Scott Balneaves <sbalneav@mate-desktop.org>
Date: Wed, 01 May 2013 14:15:58 +0000
Subject: Wrapper for Issue #9

---
diff --git a/Makefile.am b/Makefile.am
index 3723380..2622b71 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -22,6 +22,7 @@ EXTRA_DIST = 			\
 	mate-doc-utils.make	\
 	xmldocs.make		\
 	omf.make		\
+	mate-terminal.wrapper   \
 	$(man_MANS) 	\
 	$(NULL)
 
diff --git a/mate-terminal.wrapper b/mate-terminal.wrapper
new file mode 100755
index 0000000..4d528c9
--- a/dev/null
+++ b/mate-terminal.wrapper
@@ -0,0 +1,44 @@
+#!/usr/bin/env python
+
+import sys
+import os
+import subprocess
+
+newargs = ['mate-terminal']
+oldargs = sys.argv[1:]
+
+while True:
+    try:
+        arg = oldargs.pop(0)
+
+        if arg == '-display':
+            os.environ['DISPLAY'] = oldargs.pop(0)
+        elif arg == '-name':
+            newargs.append('--window-with-profile=' + oldargs.pop(0))
+        elif arg == '-n':
+            sys.stderr.write('Set an icon in your profile')
+        elif arg == '-T' or arg == '-title':
+            newargs.append('-t')
+            newargs.append(oldargs.pop(0))
+        elif arg == '-ls' or arg == '+ls':
+            sys.stderr.write('Login shell not supported.  Set in your profile.')
+        elif arg == '-geometry':
+            newargs.append('--geometry=' + oldargs.pop(0))
+        elif arg == '-fn':
+            newargs.append('--font=' + oldargs.pop(0))
+        elif arg == '-fg':
+            newargs.append('--foreground=' + oldargs.pop(0))
+        elif arg == '-bg':
+            newargs.append('--background=' + oldargs.pop(0))
+        elif arg == '-tn':
+            newargs.append('--termname=' + oldargs.pop(0))
+        elif arg == '-h' or arg == '--help':
+            newargs.append('--help')
+        elif arg == '-e':
+            newargs.append('-x')
+            newargs += oldargs
+            break
+    except IndexError:
+        break
+
+subprocess.call(newargs)
--
cgit 
