From d447ebdf19cc56d601c5ef2c46a55f301ecbeda6 Mon Sep 17 00:00:00 2001
From: Scott Balneaves <sbalneav@mate-desktop.org>
Date: Thu, 02 May 2013 15:35:16 +0000
Subject: Add wrapper install

---
diff --git a/Makefile.am b/Makefile.am
index 2622b71..abe2e10 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -15,6 +15,8 @@ desktop_DATA = $(desktop_in_files:.desktop.in.in=.desktop)
 
 man_MANS = mate-terminal.1
 
+bin_SCRIPTS = mate-terminal.wrapper
+
 EXTRA_DIST = 			\
 	ChangeLog.pre-2-23	\
 	ChangeLog.README	\
--
cgit 
