From dc37adbba5f85bd968081ff2db18c5a5f9014162 Mon Sep 17 00:00:00 2001
From: Stefano Karapetsas <stefano@karapetsas.com>
Date: Wed, 03 Apr 2013 15:16:32 +0000
Subject: Add support for Compiz 0.9.x

---
diff --git a/capplets/appearance/appearance-support.c b/capplets/appearance/appearance-support.c
index 220e8cf..42b906c 100644
--- a/capplets/appearance/appearance-support.c
+++ b/capplets/appearance/appearance-support.c
@@ -48,6 +48,7 @@ metacity_is_running()
     current_wm = wm_common_get_current_window_manager ();
 
     is_running = (g_strcmp0(current_wm, WM_COMMON_METACITY) == 0) ||
+                 (g_strcmp0(current_wm, WM_COMMON_COMPIZ_OLD) == 0) ||
                  (g_strcmp0(current_wm, WM_COMMON_COMPIZ) == 0);
 
     g_free (current_wm);
diff --git a/capplets/common/wm-common.h b/capplets/common/wm-common.h
index baa51d3..6c41f9b 100644
--- a/capplets/common/wm-common.h
+++ b/capplets/common/wm-common.h
@@ -1,11 +1,12 @@
 #ifndef WM_COMMON_H
 #define WM_COMMON_H
 
-#define WM_COMMON_MARCO    "Marco"
-#define WM_COMMON_SAWFISH  "Sawfish"
-#define WM_COMMON_METACITY "Metacity"
-#define WM_COMMON_COMPIZ   "compiz"
-#define WM_COMMON_UNKNOWN  "Unknown"
+#define WM_COMMON_MARCO       "Marco"
+#define WM_COMMON_SAWFISH     "Sawfish"
+#define WM_COMMON_METACITY    "Metacity"
+#define WM_COMMON_COMPIZ      "Compiz"
+#define WM_COMMON_COMPIZ_OLD  "compiz"
+#define WM_COMMON_UNKNOWN     "Unknown"
 
 gchar *wm_common_get_current_window_manager (void);
 /* Returns a strv of keybinding names for the window manager;
--
cgit 
