Description: Fix building on systems with Linux >= 4.7
Author: Martin Wimpress <code@flexion.org>

diff -rupN a/cpufreq/src/cpufreq-monitor-libcpufreq.c b/cpufreq/src/cpufreq-monitor-libcpufreq.c
--- a/cpufreq/src/cpufreq-monitor-libcpufreq.c
+++ b/cpufreq/src/cpufreq-monitor-libcpufreq.c
@@ -111,11 +111,7 @@ cpufreq_monitor_libcpufreq_run (CPUFreqM
 		/* Check whether it failed because
 		 * cpu is not online.
 		 */
-#if LINUX_VERSION_CODE < KERNEL_VERSION(4, 7, 0)
 		if (!cpufreq_cpu_exists (cpu)) {
-#else
-		if (cpupower_is_cpu_online (cpu)) {
-#endif
 			g_object_set (G_OBJECT (monitor), "online", FALSE, NULL);
 			return TRUE;
 		}
