From f4a927c268231fdd6b5a333d0c547b46a11af858 Mon Sep 17 00:00:00 2001
From: Stefano Karapetsas <stefano@karapetsas.com>
Date: Mon, 08 Apr 2013 14:07:33 +0000
Subject: clock: Allow users to set custom format in dconf-editor

---
diff --git a/applets/clock/clock.c b/applets/clock/clock.c
index 44b844e..bcbf847 100644
--- a/applets/clock/clock.c
+++ b/applets/clock/clock.c
@@ -2264,6 +2264,17 @@ static void
 setup_gsettings (ClockData *cd)
 {
         cd->settings = mate_panel_applet_settings_new (MATE_PANEL_APPLET (cd->applet), CLOCK_SCHEMA);
+
+        /* hack to allow users to set custom format in dconf-editor */
+        gint format;
+        gchar *custom_format;
+        format = g_settings_get_enum (cd->settings, KEY_FORMAT);
+        custom_format = g_settings_get_string (cd->settings, KEY_CUSTOM_FORMAT);
+        g_settings_set_enum (cd->settings, KEY_FORMAT, format);
+        g_settings_set_string (cd->settings, KEY_CUSTOM_FORMAT, custom_format);
+        if (custom_format != NULL)
+                g_free (custom_format);
+
         g_signal_connect (cd->settings, "changed::" KEY_FORMAT, G_CALLBACK (format_changed), cd);
         g_signal_connect (cd->settings, "changed::" KEY_SHOW_SECONDS, G_CALLBACK (show_seconds_changed), cd);
         g_signal_connect (cd->settings, "changed::" KEY_SHOW_DATE, G_CALLBACK (show_date_changed), cd);
--
cgit 
