From d6cc1de269ca4fa4404137d075d6e59fc20dc4fb Mon Sep 17 00:00:00 2001
From: Stefano Karapetsas <stefano@karapetsas.com>
Date: Sun, 07 Apr 2013 18:48:15 +0000
Subject: Fix window screenshot keybinding

---
diff --git a/src/include/all-keybindings.h b/src/include/all-keybindings.h
index 26f2326..10488a1 100644
--- a/src/include/all-keybindings.h
+++ b/src/include/all-keybindings.h
@@ -184,7 +184,7 @@ keybind (run-command-31, handle_run_command, 30, 0)
 keybind (run-command-32, handle_run_command, 31, 0)
 
 keybind (run-command-screenshot, handle_run_command, 32, 0)
-keybind (run-command-window_screenshot, handle_run_command, 33, 0)
+keybind (run-command-window-screenshot, handle_run_command, 33, 0)
 
 keybind (run-command-terminal, handle_run_terminal, 0, 0)
 
--
cgit 
