From 2c5525c07a233d39d3bb50bf55aff3c611a84e3b Mon Sep 17 00:00:00 2001
From: Stefano Karapetsas <stefano@karapetsas.com>
Date: Sat, 26 May 2012 12:26:55 +0200
Subject: [PATCH] move is_program_in_path() functon

---
 src/gs-lock-plug.c |   30 +++++++++++++++---------------
 1 file changed, 15 insertions(+), 15 deletions(-)

diff --git a/src/gs-lock-plug.c b/src/gs-lock-plug.c
index daf2632..70cf1d7 100644
--- a/src/gs-lock-plug.c
+++ b/src/gs-lock-plug.c
@@ -172,6 +172,21 @@ gs_lock_plug_style_set (GtkWidget *widget,
 	gtk_box_set_spacing (GTK_BOX (plug->priv->auth_action_area), 5);
 }
 
+static gboolean
+is_program_in_path (const char *program)
+{
+	char *tmp = g_find_program_in_path (program);
+	if (tmp != NULL)
+	{
+		g_free (tmp);
+		return TRUE;
+	}
+	else
+	{
+		return FALSE;
+	}
+}
+
 static void
 do_user_switch (GSLockPlug *plug)
 {
@@ -917,21 +932,6 @@ gs_lock_plug_set_logout_command (GSLockPlug *plug,
 	}
 }
 
-static gboolean
-is_program_in_path (const char *program)
-{
-	char *tmp = g_find_program_in_path (program);
-	if (tmp != NULL)
-	{
-		g_free (tmp);
-		return TRUE;
-	}
-	else
-	{
-		return FALSE;
-	}
-}
-
 static void
 gs_lock_plug_set_status_message (GSLockPlug *plug,
                                  const char *status_message)
-- 
1.7.10

