From a7a7d80dff2f26c8dfbc077be5c8e15fe341406f Mon Sep 17 00:00:00 2001
From: Stefano Karapetsas <stefano@karapetsas.com>
Date: Tue, 02 Apr 2013 15:05:50 +0000
Subject: fix for extra quotes getting passed to mate-bluetooth-sendto via mate-bluetooth-applet

https://bugs.launchpad.net/linuxmint/+bug/1005649/comments/35
---
diff --git a/applet/main.c b/applet/main.c
index daf6672..f582003 100644
--- a/applet/main.c
+++ b/applet/main.c
@@ -207,13 +207,13 @@ void sendto_callback(GObject *widget, gpointer user_data)
 	if (address != NULL) {
 		char *s;
 
-		s = g_strdup_printf ("--device=\"%s\"", address);
+		s = g_strdup_printf ("--device=%s", address);
 		g_ptr_array_add (a, s);
 	}
 	if (address != NULL && alias != NULL) {
 		char *s;
 
-		s = g_strdup_printf ("--name=\"%s\"", alias);
+		s = g_strdup_printf ("--name=%s", alias);
 		g_ptr_array_add (a, s);
 	}
 	g_ptr_array_add (a, NULL);
--
cgit 
