From 2bc333f790580a6279b1817270d1fc634da7decf Mon Sep 17 00:00:00 2001
From: Dan Bravender <dan.bravender@gmail.com>
Date: Sun, 14 Apr 2013 16:37:11 +0000
Subject: Fixes #94 - default for target_navigation needs to be FALSE for spatial mode to work

---
diff --git a/src/caja-window-manage-views.c b/src/caja-window-manage-views.c
index 02b7fac..c2c5b5f 100644
--- a/src/caja-window-manage-views.c
+++ b/src/caja-window-manage-views.c
@@ -527,7 +527,7 @@ caja_window_slot_open_location_full (CajaWindowSlot *slot,
     char *old_uri, *new_uri;
     int new_slot_position;
     GList *l;
-    gboolean target_navigation, target_same;
+    gboolean target_navigation = FALSE, target_same = FALSE;
     gboolean is_desktop;
 
     window = slot->pane->window;
--
cgit 
