From d116bf8ba8be698869fef81b30d576bcacf3234c Mon Sep 17 00:00:00 2001
From: Stefano Karapetsas <stefano@karapetsas.com>
Date: Mon, 29 Apr 2013 10:41:32 +0000
Subject: Fix for x-caja-desktop windows at login

Use --no-default-window in Exec field
---
diff --git a/data/caja.desktop.in.in b/data/caja.desktop.in.in
index 8e19a81..c975198 100644
--- a/data/caja.desktop.in.in
+++ b/data/caja.desktop.in.in
@@ -1,7 +1,7 @@
 [Desktop Entry]
 _Name=Caja
 _GenericName=File Manager
-Exec=caja
+Exec=caja -n
 Icon=system-file-manager
 Terminal=false
 Type=Application
--
cgit 
