From 2ba883533e92099322f595085d7ce4941226913f Mon Sep 17 00:00:00 2001
From: Scott Balneaves <sbalneav@mate-desktop.org>
Date: Wed, 17 Apr 2013 14:54:23 +0000
Subject: Possible fix for Issue #111

---
diff --git a/libcaja-private/caja-file.c b/libcaja-private/caja-file.c
index 1958faf..6c252e3 100644
--- a/libcaja-private/caja-file.c
+++ b/libcaja-private/caja-file.c
@@ -7158,6 +7158,10 @@ caja_file_is_binary (CajaFile *file)
 	 */
 	
 	fp = g_fopen (g_file_get_path (caja_file_get_location (file)), "r");
+	if (fp == NULL)
+	{
+		return FALSE;
+	}
 	for (i = 0; i < 4096; i++) {
 		c = fgetc(fp);
 		if (c == EOF) {
--
cgit 
