From d92c2754681b24ed5afcbb0d9458297e6101254e Mon Sep 17 00:00:00 2001
From: Scott Balneaves <sbalneav@mate-desktop.org>
Date: Wed, 08 May 2013 01:24:00 +0000
Subject: Fix for #111

---
diff --git a/libcaja-private/caja-file.c b/libcaja-private/caja-file.c
index 6c252e3..84d799f 100644
--- a/libcaja-private/caja-file.c
+++ b/libcaja-private/caja-file.c
@@ -7149,7 +7149,7 @@ caja_file_is_binary (CajaFile *file)
 	
 	gboolean is_binary = FALSE;
 	int c;
-	int i;
+	int i = 0;
 	FILE *fp;
 	
 	/* Check the first 4096 bytes of the files. If these contains a 0,
@@ -7162,15 +7162,17 @@ caja_file_is_binary (CajaFile *file)
 	{
 		return FALSE;
 	}
-	for (i = 0; i < 4096; i++) {
-		c = fgetc(fp);
-		if (c == EOF) {
+	
+	while (!feof (fp)) {
+		if (i > 4096) {
 			break;
 		}
-		else if (c == 0) {
+		c = fgetc(fp);
+		if (c == 0) {
 			is_binary = TRUE;
 			break;
 		}
+		i++;
 	}
 	fclose(fp);
 	
--
cgit 
