From c79f320aba836847f35e5e465754b1ddd62806b4 Mon Sep 17 00:00:00 2001
From: Stefano Karapetsas <stefano@karapetsas.com>
Date: Sun, 22 Sep 2013 15:01:18 +0200
Subject: [PATCH] Add check if mimetype is null before apply thumbnail frame

To avoid segfaults on file deletions
---
 libcaja-private/caja-file.c | 13 +++++++------
 1 file changed, 7 insertions(+), 6 deletions(-)

diff --git a/libcaja-private/caja-file.c b/libcaja-private/caja-file.c
index 85f8a9c..fa00ac8 100644
--- a/libcaja-private/caja-file.c
+++ b/libcaja-private/caja-file.c
@@ -4324,12 +4324,13 @@ caja_file_get_icon (CajaFile *file,
 
 			/* Render frames only for thumbnails of non-image files 
 			   and for images with no alpha channel. */ 
-			gboolean is_image = strncmp(eel_ref_str_peek (file->details->mime_type), "image/", 6) == 0;
-     			if (!is_image || 
-     			    is_image && !gdk_pixbuf_get_has_alpha (raw_pixbuf)) {
-				caja_ui_frame_image (&scaled_pixbuf);
-     			}
-     			
+			gboolean is_image = file->details->mime_type &&
+				(strncmp(eel_ref_str_peek (file->details->mime_type), "image/", 6) == 0);
+				if (!is_image ||
+					is_image && !gdk_pixbuf_get_has_alpha (raw_pixbuf)) {
+					caja_ui_frame_image (&scaled_pixbuf);
+				}
+
 			g_object_unref (raw_pixbuf);
 
 			/* Don't scale up if more than 25%, then read the original
-- 
1.8.4.rc3

