diff --git a/eel/eel-glib-extensions.c b/eel/eel-glib-extensions.c
index ee10805..96eeb35 100644
--- a/eel/eel-glib-extensions.c
+++ b/eel/eel-glib-extensions.c
@@ -378,6 +378,34 @@ eel_g_str_list_index (GList *str_list,
 }
 
 /**
+ * eel_g_list_free_deep_custom
+ *
+ * Frees the elements of a list and then the list, using a custom free function.
+ *
+ * @list: List of elements that can be freed with the provided free function.
+ * @element_free_func: function to call with the data pointer and user_data to free it.
+ * @user_data: User data to pass to element_free_func
+ **/
+void
+eel_g_list_free_deep_custom (GList *list, GFunc element_free_func, gpointer user_data)
+{
+	g_list_foreach (list, element_free_func, user_data);
+	g_list_free (list);
+}
+
+/**
+ * eel_g_list_free_deep
+ *
+ * Frees the elements of a list and then the list.
+ * @list: List of elements that can be freed with g_free.
+ **/
+void
+eel_g_list_free_deep (GList *list)
+{
+	eel_g_list_free_deep_custom (list, (GFunc) g_free, NULL);
+}
+
+/**
  * eel_g_strv_find
  *
  * Get index of string in array of strings.
diff --git a/eel/eel-glib-extensions.h b/eel/eel-glib-extensions.h
index 907a1ec..9c62c57 100644
--- a/eel/eel-glib-extensions.h
+++ b/eel/eel-glib-extensions.h
@@ -54,6 +54,12 @@ GList *     eel_g_list_partition                        (GList                 *
         gpointer               user_data,
         GList                **removed);
 
+/* List functions for lists of g_free'able objects. */
+void        eel_g_list_free_deep                        (GList                 *list);
+void        eel_g_list_free_deep_custom                 (GList                 *list,
+                                                         GFunc                  element_free_func,
+                                                         gpointer               user_data);
+
 /* List functions for lists of C strings. */
 gboolean    eel_g_str_list_equal                        (GList                 *str_list_a,
         GList                 *str_list_b);
diff --git a/src/file-manager/fm-properties-window.c b/src/file-manager/fm-properties-window.c
index 854de86..983cfe3 100644
--- a/src/file-manager/fm-properties-window.c
+++ b/src/file-manager/fm-properties-window.c
@@ -3423,7 +3423,7 @@ get_initial_emblems (GList *files)
 	ret = g_hash_table_new_full (g_direct_hash,
 				     g_direct_equal,
 				     NULL,
-				     (GDestroyNotify) g_free);
+				     (GDestroyNotify) eel_g_list_free_deep);
 
 	for (l = files; l != NULL; l = l->next) {
 		CajaFile *file;
