From 33ac2c8e15a6dbf22982fd469af1446d210eb2c8 Mon Sep 17 00:00:00 2001
From: Stefano Karapetsas <stefano@karapetsas.com>
Date: Sat, 06 Apr 2013 11:09:37 +0000
Subject: Init libnotify in mate-settings-daemon, not in plugins

---
diff --git a/mate-settings-daemon/Makefile.am b/mate-settings-daemon/Makefile.am
index fcf4520..778cb92 100644
--- a/mate-settings-daemon/Makefile.am
+++ b/mate-settings-daemon/Makefile.am
@@ -8,6 +8,7 @@ INCLUDES = \
 	$(WARN_CFLAGS)						\
 	$(DISABLE_DEPRECATED_CFLAGS)				\
 	$(SETTINGS_DAEMON_CFLAGS)				\
+	$(LIBNOTIFY_CFLAGS)						\
 	$(NULL)
 
 noinst_LTLIBRARIES = 			\
@@ -78,7 +79,8 @@ mate_settings_daemon_LDFLAGS = 	\
 
 mate_settings_daemon_LDADD = 		\
 	libmsd-profile.la		\
-	$(SETTINGS_DAEMON_LIBS)
+	$(SETTINGS_DAEMON_LIBS)	\
+	$(LIBNOTIFY_LIBS)
 
 EXTRA_DIST = 				\
 	$(Dbusapi_DATA)			\
diff --git a/mate-settings-daemon/main.c b/mate-settings-daemon/main.c
index aba4285..364ff75 100644
--- a/mate-settings-daemon/main.c
+++ b/mate-settings-daemon/main.c
@@ -35,6 +35,10 @@
 #include <dbus/dbus-glib.h>
 #include <dbus/dbus-glib-lowlevel.h>
 
+#ifdef HAVE_LIBNOTIFY
+#include <libnotify/notify.h>
+#endif /* HAVE_LIBNOTIFY */
+
 #include "mate-settings-manager.h"
 #include "mate-settings-profile.h"
 
@@ -483,6 +487,10 @@ main (int argc, char *argv[])
                 goto out;
         }
 
+#ifdef HAVE_LIBNOTIFY
+        notify_init ("mate-settings-daemon");
+#endif /* HAVE_LIBNOTIFY */
+
         mate_settings_profile_start ("mate_settings_manager_new");
         manager = mate_settings_manager_new ();
         mate_settings_profile_end ("mate_settings_manager_new");
@@ -521,6 +529,11 @@ main (int argc, char *argv[])
                 g_object_unref (manager);
         }
 
+#ifdef HAVE_LIBNOTIFY
+        if (notify_is_initted ())
+            notify_uninit ();
+#endif /* HAVE_LIBNOTIFY */
+
         g_debug ("SettingsDaemon finished");
         mate_settings_profile_end (NULL);
 
diff --git a/plugins/a11y-keyboard/msd-a11y-keyboard-manager.c b/plugins/a11y-keyboard/msd-a11y-keyboard-manager.c
index 32a317b..8151938 100644
--- a/plugins/a11y-keyboard/msd-a11y-keyboard-manager.c
+++ b/plugins/a11y-keyboard/msd-a11y-keyboard-manager.c
@@ -1245,10 +1245,6 @@ static void
 msd_a11y_keyboard_manager_init (MsdA11yKeyboardManager *manager)
 {
         manager->priv = MSD_A11Y_KEYBOARD_MANAGER_GET_PRIVATE (manager);
-
-#ifdef HAVE_LIBNOTIFY
-        notify_init ("mate-settings-daemon");
-#endif /* HAVE_LIBNOTIFY */
 }
 
 static void
--
cgit 
