From d4405221511fd45d8ba87fc75971db390391fbda Mon Sep 17 00:00:00 2001
From: Stefano Karapetsas <stefano@karapetsas.com>
Date: Sun, 7 Apr 2013 11:43:32 +0200
Subject: [PATCH] Fix MateConf convert file

---
 data/eom.convert |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/data/eom.convert b/data/eom.convert
index dcd2b02..214026e 100644
--- a/data/eom.convert
+++ b/data/eom.convert
@@ -1,4 +1,4 @@
-[org.mate.eom.fullscreen]
+[org.mate.eom.full-screen]
 loop=/apps/eom/full_screen/loop
 seconds=/apps/eom/full_screen/seconds
 upscale=/apps/eom/full_screen/upscale
-- 
1.7.10.4

